### PR TITLE
continuing documentation and cleanup of map_object

### PR DESCRIPTION
**Pull request by red031000**

### CODE REVIEW

**Comment by red031000** [comment]
ready for review

File: src/map_object.c [hunk 1]
```diff
@@ -287,7 +374,7 @@ void sub_0205E580(MapObjectManager *manager) {
     LocalMapObject *objects = MapObjectManager_GetObjects(manager);
 
     do {
-        if (MapObject_IsInUse(objects) == TRUE && MapObject_GetID(objects) == 0xFF) { // todo: const (is this invalid ID? or just max?)
+        if (MapObject_CheckActive(objects) == TRUE && MapObject_GetID(objects) == 0xFF) { // todo: const (is this invalid ID? or just max?)
```
Comment (tgsm) [inline src/map_object.c:377]: This is probably `obj_player`

Comment (red031000) [reply src/map_object.c:377]: possible, however I would rather have either an existing header for consts, or if there's a new one needed, valued for the other consts
also I'm not 100% sure about it, and I would like to be

**Review by tgsm — APPROVED** [review]
I have no major complaints - thanks 👍

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**